src/babylon.scene.ts

```diff
@@ -3220,6 +3232,7 @@
          */
         public addMaterial(newMaterial: Material): void {
             this.materials.push(newMaterial);
+            this.onNewMaterialAddedObservable.notifyObservers(newMesh);

```

Comment:
I'm fine with these observables. that's a good idea.  You only need to clear them at dispose time

Reply:
Hello please do it asap because we are actually in code freeze and I should not accept any more changes for 3.3. But as you started the PR the day before I'm doing an exception.
Regarding co we are good but the content of your comments is wrong

besides that we can merge when you want. I'll keep this PR open for now waiting for the additional observables

Reply:
I did it as soon as I can in a single commit c900ceae4819e07083fa2e69abad1584c7e929b3.
I hope I didn't forget places to notify observers.